### DISCUSSION TITLE
Voice playback events

### DISCUSSION THREAD

**Discussion by GordiNoki** [shardlab/discordrb#165] [Q&A]
I'm trying to make queue based music bot, but I can't find any way to detect current file play ending.

**Comment by Daniel-Worrall** [comment]
Can you describe further on what you're trying to accomplish, what you have tried so far, and what you currently have?

**Reply by GordiNoki** [reply]
I made command parser and per guild data store, so I can get queue and all other needed data. When I write a command, bot connects to VC and plays current song. But I cannot find any way to actually detect ending of current track, so I can process queue and play next track.

**Reply by Daniel-Worrall** [reply]
What method are you using the play the song? It should have blocking behaviour for the duration of the song, so you can call the next song after it

**Reply by GordiNoki** [reply]
I'm using VoiceBot.play_file(url) method, is there a way to block thread with this one?

**Reply by Daniel-Worrall** [reply]
No such method exists. Did you define it yourself?

**Reply by GordiNoki** [reply]
Oh, play_file, sorry

**Reply by GordiNoki** [reply]
Wait, I'm stupid, thanks for directing to solution

**Reply by Droid00000** [reply]
Glad you were able to find a solution!
